modules/core/src/main/java/org/apache/synapse/core/axis2/Axis2FlexibleMEPClient.java

```diff
@@ -159,6 +160,15 @@ public static void send(
         String preserveAddressingProperty = (String) synapseOutMessageContext.getProperty(
                 SynapseConstants.PRESERVE_WS_ADDRESSING);
         MessageContext axisOutMsgCtx = cloneForSend(originalInMsgCtx, preserveAddressingProperty);
+
+        // set "INTERNAL_TRANSACTION_COUNTED" property in axisOutMsgCtx's message context, if it is present in the message
+        // context of the originalInMsgCtx
+        Object transactionProperty = originalInMsgCtx.getProperty(BaseConstants.INTERNAL_TRANSACTION_COUNTED);
+        if (transactionProperty instanceof Boolean) {
+            axisOutMsgCtx.setProperty(BaseConstants.INTERNAL_TRANSACTION_COUNTED,
+                                      Boolean.getBoolean(transactionProperty.toString()));
```

Comment:
are we setting the INTERNAL_TRANSACTION_COUNTED value as a System property?

Reply:
No, we are not.